src/strategy/weekly.rs

```diff
@@ -0,0 +1,122 @@
+//! Strategy for weekly weekly updates.
+
+use crate::config::inputs;
+use crate::weekly::{utils, WeeklyCalendar, WeeklyWindow};
+use failure::{Error, Fallible};
+use futures::future;
+use futures::prelude::*;
+use log::trace;
+use serde::Serialize;
+use std::pin::Pin;
+use std::time::Duration;
+
+/// Strategy for weekly weekly updates.
+#[derive(Clone, Debug, Default, Serialize)]
+pub(crate) struct StrategyWeekly {
+    /// Whitelisted time windows during which updates are allowed.
+    allowed_intervals: WeeklyCalendar,
```

Comment:
How about calling this e.g. `schedule`? It feels funny to name it something that refers to the underlying implementation when the struct name is meant to abstract it.